src/hugr/rewrite/replace.rs

```diff
@@ -482,7 +482,7 @@ mod test {
             FunctionType::new_linear(just_list.clone()).with_extension_delta(&exset),
         )?;
 
-        let pred_const = cfg.add_constant(ops::Const::unary_unit_sum(), None)?;
+        let pred_const = cfg.add_constant(ops::Const::unary_unit_sum(), exset)?;
```

Comment:
This is because we can no longer infer an extension set for the Const itself, right? Presumably we could still use None for the LoadConstant? So what about changing the builder to automatically use `pure` for the Const node, but still allow specifying something for the LoadConstant?

Reply:
Ah, no, hang on - this is just the Const, there is no LoadConstant here. So IOW, the `exset` could be anything, that will play no further part in inference?

Hmmm, we should drop the param. That could be another PR/issue, tho.

Reply:
I think we should drop the parameter, default to `pure` for const and move this param to `load_constant`. I'll open another PR for this.
At the moment though, `load_constant` uses the extension set that is provided with the ConstId, which is why `exset